apt-avahi-discover

```diff
@@ -41,22 +41,16 @@ def get_avahi_discover_timeout():
 
 
 @functools.total_ordering
-class AptAvahiClient(asyncore.dispatcher):
+class AptAvahiClient:
     def __init__(self, addr):
-        asyncore.dispatcher.__init__(self)
-        if is_ipv6(addr[0]):
-            self.create_socket(socket.AF_INET6, socket.SOCK_STREAM)
-            self.connect((addr[0], addr[1], 0, 0))
-        else:
-            self.create_socket(socket.AF_INET, socket.SOCK_STREAM)
-            self.connect(addr)
-        self._time_init = time.time()
         self.time_to_connect = sys.maxsize
         self.address = addr
 
-    def handle_connect(self):
-        self.time_to_connect = time.time() - self._time_init
-        self.close()
+    async def time_connect(self):
+        _time_init = time.time()
+        reader, writer = await asyncio.open_connection(self.address[0], self.address[1])
+        self.time_to_connect = time.time() - _time_init
+        writer.close()
```

Comment:
Silly question - why is only the `writer` closed here but not the reader?

Reply:
There is no `close` method in the `StreamReader` class. My understanding is that calling `close` on the associated `StreamWriter` closes the entire transport (socket) rather than leaving the socket in a half-closed state. This is also how the `asyncio` examples were written (https://docs.python.org/3/library/asyncio-stream.html#examples).

Reply:
Thanks, makes sense!